src/proto/media/file/regular.rs

```diff
@@ -21,31 +21,54 @@ impl RegularFile {
         Self(handle)
     }
 
-    /// Read data from file
+    /// Read data from a file, hence, a regular file or a directory.
     ///
-    /// Try to read as much as possible into `buffer`. Returns the number of bytes that were
-    /// actually read.
+    /// # Read from Regular Files
+    /// If `self` is not a directory, the function reads the requested number of bytes from the file
+    /// at the file’s current position and returns them in `buffer`. If the read goes beyond the end
+    /// of the file, the read length is truncated to the end of the file. The file’s current
+    /// position is increased by the number of bytes returned.
+    ///
+    /// # Read from Directory
+    /// If `self` is a directory, the function reads the directory entry at the file’s current
+    /// position and returns the entry in `buffer`. If the `buffer` is not large enough to hold the
+    /// current directory entry, then `EFI_BUFFER_TOO_SMALL` is returned and the current file
+    /// position is not updated. `buffer_size` is set to be the size of the buffer needed to read
+    /// the entry. On success, the current position is updated to the next directory entry. If there
+    /// are no more directory entries, the read returns a zero-length buffer. `EFI_FILE_INFO` is the
+    /// structure returned as the directory entry.
     ///
     /// # Arguments
     /// * `buffer`  The target buffer of the read operation
     ///
-    /// # Errors
+    /// # Return Value
+    /// This either returns the size of read bytes in the success case or a tuple of the UEFI
+    /// status and the required buffer size, if the buffer was too small. The status code can
+    /// signalize one of the following errors:
+    ///
+    /// ## Status Error Codes
     /// * `uefi::Status::NO_MEDIA`           The device has no media
     /// * `uefi::Status::DEVICE_ERROR`       The device reported an error, the file was deleted,
     ///                                      or the end of the file was reached before the `read()`.
     /// * `uefi::Status::VOLUME_CORRUPTED`   The filesystem structures are corrupted
     /// * `uefi::Status::BUFFER_TOO_SMALL`   The buffer is too small to hold a directory entry,
     ///                                      and the required buffer size is provided as output.
-    pub fn read(&mut self, buffer: &mut [u8]) -> Result<usize, Option<usize>> {
+    pub fn read(&mut self, buffer: &mut [u8]) -> Result<usize, (Status, Option<usize>)> {
```

Comment:
I think returning the `Status` should not be needed, since that information is already in the `uefi::Result` type alias: https://docs.rs/uefi/latest/uefi/type.Result.html (see `status` method)

Reply:
Yikes, I didn't see. Thanks! I updated the MR and the documentation to better reflect this.